tests/brain/test_brain.py

```diff
@@ -1705,7 +1705,8 @@ def test_infer_dict_from_keys() -> None:
     )
     for node in bad_nodes:
         with pytest.raises(InferenceError):
-            next(node.infer())
+            if isinstance(next(node.infer()), util.UninferableBase):
```

Comment:
Why is this necessary?

Reply:
For this one, I don't know exactly, but my intuition was the following. Before, when we had `Unknown` for a parent, when inference encountered Unknown, it would raise (`InferenceError`). Now, there is no `Unknown`, but the calls still aren't correct. So, we just fail to infer and yield `Uninferable`. 

However, both responses kind of seem to make sense, so I opted to allow either of the responses (`InferenceError` or `Uninferable`) to pass the tests.

Reply:
Can we be certain that it is either of the two? So, just assert that it is always `Uniferable` or always the exception? If so, I'd prefer just testing for that

Reply:
Yes, right now it's always `Uninferable`. I'll change to test specifically for that.

Reply:
fixed

Reply:
Ah, one test fails under `pypy` with `InferenceError`. I'd rather go with the initial `pytest.raises` clause then